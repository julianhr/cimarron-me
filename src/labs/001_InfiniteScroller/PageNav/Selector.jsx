import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'


const Root = styled.div`
  padding: 20px 0 40px;
  display: flex;
  flex-direction: column;

  label {
    display: block;
    padding-bottom: 5px;
  }

  select {
    background-color: ${({ theme }) => theme.colors.field.background};
  }
`

function Selector({ label, keys, titles, selected, onChange }) {
  const renderOptions = () => {
    return keys.map((key, i) => (
      <option
        key={key}
        value={key.toString()}
      >
        {titles[i]}
      </option>
    ))
  }

  return (
    <Root>
      <label htmlFor='scroller-type'>{label}</label>
      <select
        id='scroller-type'
        value={selected}
        onChange={(event) => onChange(event.target.value)}
      >
        {renderOptions()}
      </select>
    </Root>
  )
}

Selector.propTypes = {
  label: PropTypes.string.isRequired,
  keys: PropTypes.array.isRequired,
  titles: PropTypes.array.isRequired,
  selected: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func.isRequired,
}

export default Selector
