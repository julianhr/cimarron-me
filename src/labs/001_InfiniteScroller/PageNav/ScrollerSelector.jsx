import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { setScrollerType } from '../actions/rootActions'
import Selector from './Selector'


export const SCROLLERS = [
  { intersectionObserver: 'Intersection Observer with Sentinel' },
  { containerScrollHeights: 'Container scrollHeight and scrollTop' },
  { sentinelClientRect: 'Sentinel bounding client rectangle' },
]

export function ScrollerSelector({ scrollerType, setScrollerType }) {
  return (
    <Selector
      label='Scroller Re-Fetch Method:'
      keys={SCROLLERS.map(scroller => Object.keys(scroller))}
      titles={SCROLLERS.map(scroller => Object.values(scroller))}
      selected={scrollerType}
      onChange={setScrollerType}
    />
  )
}

ScrollerSelector.propTypes = {
  setScrollerType: PropTypes.func.isRequired,
  scrollerType: PropTypes.string.isRequired,
}

const mapStateToProps = ({ scrollerType }) => ({ scrollerType })
const mapDispatchToProps = { setScrollerType }

export default connect(mapStateToProps, mapDispatchToProps)(ScrollerSelector)
