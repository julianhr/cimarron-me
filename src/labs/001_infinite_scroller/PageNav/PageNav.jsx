import React from 'react'
import styled from '@emotion/styled'

import ScrollerSelector from './ScrollerSelector'


const Root = styled.nav`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;

  ${props => props.theme.queries.from('md')} {
    padding-top: 30px;
  }
`

const H2 = styled.h2`
  font-size: 30px;
`

class TopNav extends React.PureComponent {
  render() {
    return (
      <Root>
        <H2>Infinite Scroller</H2>
        <ScrollerSelector />
      </Root>
    )
  }
}

export default TopNav
