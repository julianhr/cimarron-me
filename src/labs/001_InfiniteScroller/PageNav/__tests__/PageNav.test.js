import React from 'react'
import { render, mount, shallow } from 'enzyme'
import { create } from 'react-test-renderer'

import PageNav from '../PageNav'
import MockInfiniteScrollerApp from '../../__tests__/__mocks__/MockInfiniteScrollerApp'


describe('PageNav', () => {
  it('matches snapshot', () => {
    const wrapper = mount(
      <MockInfiniteScrollerApp>
        <PageNav />
      </MockInfiniteScrollerApp>
    )

    expect(wrapper).toMatchSnapshot()
    wrapper.unmount()
  })
})
