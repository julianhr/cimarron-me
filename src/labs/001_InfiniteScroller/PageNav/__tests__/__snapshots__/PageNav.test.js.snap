// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNav matches snapshot 1`] = `
Array [
  <ForwardRef(render)
    styles={[Function]}
  />,
  <Root>
    <nav
      className="css-1p2mroz-Root e8olbbk0"
    >
      <H2>
        <h2
          className="css-yu1g52-H2 e8olbbk1"
        >
          Infinite Scroller
        </h2>
      </H2>
      <ConnectFunction>
        <Root>
          <div
            className="css-1o7hb61-Root el5yv3l0"
          >
            <label
              htmlFor="scroller-type"
            >
              Scroller Re-Fetch Method:
            </label>
            <select
              id="scroller-type"
              onChange={[Function]}
              value="containerScrollHeights"
            >
              <option
                key="intersectionObserver"
                value="intersectionObserver"
              >
                Intersection Observer with Sentinel
              </option>
              <option
                key="containerScrollHeights"
                value="containerScrollHeights"
              >
                Container scrollHeight and scrollTop
              </option>
              <option
                key="sentinelClientRect"
                value="sentinelClientRect"
              >
                Sentinel bounding client rectangle
              </option>
            </select>
          </div>
        </Root>
      </ConnectFunction>
    </nav>
  </Root>,
]
`;
